### PR TITLE
Migration to go.mod for dependency management

### PR DESCRIPTION
**Pull request by cmaglie**
- use go.mod instead of `dep` for dependency management
- removed the `vendor` folder
- copied the `cli-inception` branch of arduino-builder in the `legacy/builder` folder (in the future this module will be migrated in the cli as a proper idiomatic go package)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,26 +1,18 @@
 language: go
 
 go:
-  - 1.11.x
+  - 1.12.x
 
 env:
-  - DEP_VERSION="0.5.0"
+  - GO111MODULE=on
```
Comment (masci) [inline .travis.yml]: do we need this provided that Go version is 1.12 now?

File: common/formatter/text.go [hunk 2]
```diff
@@ -22,8 +22,8 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/cmaglie/pb"
```
Comment (masci) [inline common/formatter/text.go]: the fork is diverging from upstream, any chance we can contribute instead of forking?

File: .travis.yml (hunk 1 above)
Comment (cmaglie) [reply .travis.yml]: it is needed for travis, it will be `on` by default from 1.13 AFAIK

File: common/formatter/text.go (hunk 2 above)
Comment (cmaglie) [reply common/formatter/text.go]: Upstream library went from 1.x to 2.x (with API change) I don't think this patch will be accepted.
Maybe we can port directly to 2.x. Anyway I'm not bound to this particular library, we can evaluate other options. The features I missed much are:

* the progress bar should delete itself when reaching 100% so we have a clean output
* it should have no race conditions! (especially if there is a "rendering" go routine that runs in parallel with the download)

Comment (masci) [reply common/formatter/text.go]: Thanks for the explanation, we can have a look at v2 or alternatives later I guess

**Pull request merged**